### PR TITLE
Adds java ->clj utility function

### PR DESCRIPTION
**Pull request by jpb**

### CODE REVIEW

File: src/com/unbounce/treajure/java.clj [hunk 1]
```diff
@@ -0,0 +1,20 @@
+(ns com.unbounce.treajure.java)
+
+(defprotocol ConvertibleToClojure
```
Comment (roman) [inline src/com/unbounce/treajure/java.clj]: Why not just `ToClojure`?

Comment (jpb) [inline src/com/unbounce/treajure/java.clj]: :+1: Good idea

File: src/com/unbounce/treajure/java.clj [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(ns com.unbounce.treajure.java)
+
+(defprotocol ToClojure
+  (->clj [o]))
+
+(extend-protocol ToClojure
+  java.util.Map
+  (->clj [o] (let [entries (.entrySet o)]
```
Comment (roman) [inline src/com/unbounce/treajure/java.clj]: what happens when keys are not String?

Comment (jpb) [inline src/com/unbounce/treajure/java.clj]: Are you referring to the line below with `^String`?

Comment (roman) [inline src/com/unbounce/treajure/java.clj]: Yeah, I think that doesn't enforce that k is a String, is just a casting optimization (which will fail at runtime), there are legitimate uses of keys that are not `String`

Comment (jpb) [inline src/com/unbounce/treajure/java.clj]: :+1: I'll remove it

**Pull request merged**